core/src/main/java/com/datastax/oss/driver/api/core/config/map/OptionsMap.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core.config.map;
+
+import com.datastax.oss.driver.api.core.config.DriverExecutionProfile;
+import com.datastax.oss.driver.api.core.config.DriverOption;
+import com.datastax.oss.driver.internal.core.config.map.DefaultOptionsMap;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Map;
+
+public interface OptionsMap {
+
+  @NonNull
+  static OptionsMap empty() {
+    return new DefaultOptionsMap();
+  }
+
+  @NonNull
+  static OptionsMap driverDefaults() {
+    return DefaultOptionsMap.driverDefaults();
+  }
+
+  @Nullable
+  <ValueT> ValueT put(
+      @NonNull String profile, @NonNull TypedDriverOption<ValueT> option, @NonNull ValueT value);
+
+  @Nullable
+  default <ValueT> ValueT put(@NonNull TypedDriverOption<ValueT> option, @NonNull ValueT value) {
```

Comment:
I wonder how this new API plays with @adutra question about the support of building map settings dynamically. Right now it seems that we are no longer exposing raw API  (to allow users to just add `object, object`). I see that we left the method `asRaw()` as a back-door, will this be enough?

Reply:
The config is not meant to support fully dynamic options. The driver internals use {{DriverOption}} to retrieve the values, so you need pre-existing instances for every option.

{{asRawMap()}} is not a backdoor, it's only intended to pass the options to the `DriverConfig` implementation. If I could hide it I would.

Reply:
`asRawMap()` is clunky ☹️ 
I'll try to revisit that, but it's tricky because `MapBasedDriverConfig/ExecutionProfile` does lookups by `DriverOption`, and I need a live view of the `OptionsMap`.